pytorch_lightning/trainer/trainer.py

```diff
@@ -1237,11 +1237,13 @@ def call_hook(self, hook_name: str, *args, **kwargs) -> Any:
                 hook_fx = getattr(model_ref, hook_name)
                 output = hook_fx(*args, **kwargs)
 
-            # if the PL module doesn't have the hook then call the accelerator
-            # used to auto-reduce things for the user with Results obj
```

Comment:
one example is `training_step_end` for DP which is handled in the accelerator. If user doesn't do the reduction we do it in the accelerator. But now with the change, if the user has it overridden you still call `training_step_end` on the DP plugin which will do another automatic reduction.